src/pages/apps/[id]/environments/[env-id]/deployments/runtime-logs/actions.ts

```diff
@@ -41,18 +41,20 @@ export const useFetchDeploymentRuntimeLogs = ({
     setError(undefined);
 
     api
-      .fetch<{ logs: Log[], totalPage:number }>(`/app/${appId}/logs?deploymentId=${deploymentId}&page=${page}`)
-      .then(data  => {
+      .fetch<{ logs: Log[]; totalPage: number }>(
+        `/app/${appId}/logs?deploymentId=${deploymentId}&page=${page}`
+      )
+      .then(data => {
         setLogs([...logs, ...data.logs]);
-        setTotalPage(data.totalPage)
+        setTotalPage(data.totalPage);
       })
-      .catch(() => {
+      .catch(res => {
         setError("Something went wrong while fetching logs.");
       })
       .finally(() => {
         setLoading(false);
       });
-  }, [appId, deploymentId, page, totalPage]);
```

Comment:
The values in these array cause the effect to be re-triggered when they are updated. Since we set `totalPage` inside the `useEffect` and the value will be updated, it will cause an unnecessary re-fetch (it may even cause an infinite loop if `totalPage` was changing with every fetch).